flow/record/base.py

```diff
@@ -14,10 +14,24 @@
 import warnings
 from datetime import datetime, timezone
 from itertools import zip_longest
-from typing import Any, Dict, Iterator, List, Mapping, Optional, Sequence, Tuple
+from pathlib import Path
+from typing import (
+    IO,
+    Any,
+    BinaryIO,
+    Dict,
+    Iterator,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
```

Comment:
Can you replace all occurrences of `Dict`, `List` and `Tuple` with `dict`, `list` and `tuple`?